test/java/org/opendatakit/briefcase/export/ExportToCsvScenario.java

```diff
@@ -0,0 +1,295 @@
+    createDirectories(instanceDir);
+    Path instanceFile = instanceDir.resolve("submission.xml");
+    write(instanceFile, instanceContent.getBytes());
+  }
+
+  void createOutputFile(String dir, String file) {
+    createDirectories(outputDir.resolve("old").resolve(dir));
+    write(outputDir.resolve("old").resolve(dir).resolve(file), "Some content".getBytes());
+    createDirectories(outputDir.resolve("new").resolve(dir));
+    write(outputDir.resolve("new").resolve(dir).resolve(file), "Some content".getBytes());
+  }
+
+  void assertNoOutputMediaDir() {
+    assertThat(Files.exists(outputDir.resolve("old").resolve("media")), is(false));
+    assertThat(Files.exists(outputDir.resolve("new").resolve("media")), is(false));
+  }
+
+  String readFile(String fileName) {
+    return new String(readAllBytes(getPath(fileName)));
+  }
+
+  static Path getPath(String fileName) {
+    return maybeGetPath(fileName).orElseThrow(RuntimeException::new);
+  }
+
+  private static Optional<Path> maybeGetPath(String fileName) {
+    return Optional.ofNullable(ExportToCsvTest.class.getClassLoader().getResource("org/opendatakit/briefcase/export/" + fileName))
+        .map(url -> Paths.get(toURI(url)));
+  }
+
+  private static Optional<String> readInstanceId(String formName) {
+    return maybeGetPath(formName + "_submission.xml")
+        .filter(Files::exists)
+        .map(ExportToCsvScenario::readInstanceId);
+  }
+
+  private static String readInstanceId(Path path) {
+    return readAllLines(path).get(0).split("instanceID=\"")[1].split("\"")[0];
+  }
+
+  private static Path installForm(Path formDir, final String formName) {
+    // Locate and copy the form definition
+    Path sourceForm = getPath(formName + ".xml");
+    Path targetForm = formDir.resolve("form.xml");
+    copy(sourceForm, targetForm);
+
+    // Prepare the instances directory
+    Path instancesDir = formDir.resolve("instances");
+    createDirectories(instancesDir);
+
+    // Copy a submission and possibly other files
+    maybeGetPath(formName + "_submission.xml")
+        .filter(Files::exists)
+        .ifPresent(path -> {
+          // Read the submission's instance ID
+          String instanceId = readInstanceId(path);
+
+          // Create a dir for this submission
+          Path submissionDir = instancesDir.resolve(instanceId.replace(":", ""));
+          createDirectories(submissionDir);
+
+          // We will copy every file we find that is prefixed with the
+          // name of the form we're installing, ignoring PEM files and templates
+          walk(path.getParent())
+              .filter(file -> file.getFileName().toString().startsWith(formName + "_"))
+              .filter(file -> !file.getFileName().toString().endsWith(".pem"))
+              .filter(file -> !file.getFileName().toString().endsWith(".tpl"))
+              .forEach(file -> {
+                Path target = submissionDir.resolve(file.getFileName().toString().substring(formName.length() + 1));
```

Comment:
I find this more readable. You?
```
.filter(file -> {
  String name = file.getFileName().toString();
  return name.startsWith(formName + "_") && !name.endsWith(".pem") && !name.endsWith(".tpl");
}).forEach(file -> {
```

Reply:
I'd prefer to leave them separate. I'll give those predicates a name to see if it's more clear that way.

Reply:
OK

Reply:
I like that less. The separate function is cool, but why not pass the extension in as an argument?